src/input.rs

```diff
@@ -0,0 +1,191 @@
+use std::cmp::Ordering;
+
+use wgpu::winit;
+
+#[derive(Debug, Clone, Copy, PartialEq, Default)]
+pub struct InputState {
+    pub camera_pan_ground: [f32; 2],
+    pub camera_pan_screen: [f32; 2],
+    pub camera_rotate: [f32; 2],
+    pub camera_zoom: f32,
+    pub camera_zoom_steps: i32,
+    pub camera_reset_viewport: bool,
+    pub close_requested: bool,
+    pub import_requested: bool,
+    pub window_resized: Option<winit::dpi::LogicalSize>,
+}
+
+#[derive(Debug, Default)]
+pub struct InputManager {
+    lmb_down: bool,
+    rmb_down: bool,
+    shift_down: bool,
+    input_state: InputState,
+}
+
+impl InputManager {
```

Comment:
We should definitely discuss, but I would leave it until after the first milestone. The thing is we don't currently know how it should look like yet. We'll have a better picture once we have a more complete application.

As can be seen from the `gui_captured_*` flags, the systems should know about each other "somehow". My current plan was to process gui first, get those two flags out of it, and pass them into our own input processing (the flags are present on imgui).

As for `InputManager` being monolithic, that is on purpose. We need to have all the controls defined somewhere, with clear precedence and priority, and possibly the ability to swap to different control layouts to simulate other 3d software - the input manager would be the one consuming raw events and producing application events dependent on its keymap. I was also envisioning the `InputManager` to be the insulation layer between raw windowing API and our application, so that various other components need know nothing about window events, possibly even allowing us to swap the windowing abstraction without rewriting a lot of other code. The intention was to map from `winit` to our own model (currently the `InputState` struct). The inspiration for that came from this tutorial: https://lokathor.github.io/learn-gfx-hal/01_introduction.html. I'd like to hear your thoughts on this in person.

If we are going to clean things up some way, I would first suggest to create an `Application` struct holding most of the current `main()` function's logic. Usage could look like:

```rust
// pseudocode

app.start_frame(); // reset state, counters, and whatnot
app.process_inputs(); // pipe inputs to gui and our own components
app.process_computations(); // dispatch computations to jobs, collect job results
app.draw();
```